docs/guides/migration_guide_vpcgw_v2.md

```diff
@@ -0,0 +1,167 @@
+---
+page_title: "Migrating from Legacy VPC Gateway to v2"
+---
+
+# Migrating from Legacy VPC Gateway to v2
+
+This guide explains how to migrate from the legacy VPC gateway configuration (v1) to the new v2 API.
+In the legacy setup, DHCP and DHCP reservations are managed with dedicated resources and referenced in the gateway network.
+In v2, the public gateway is migrated to use IPAM (IP Address Management) mode.
+In 2023, DHCP functionality was moved from Public Gateways to Private Networks, DHCP resources are now no longer needed.
+
+Note:
+During migration, you need to trigger the migration call by setting the `migrate_to_v2` flag on your public gateway resource.
+You can do this via the Terraform configuration or by using the Scaleway CLI/Console.
+
+## Prerequisites
+
+### Ensure the Latest Provider Version
+
+Ensure your Scaleway provider is updated to at least version `2.52.0`.
+
+```hcl
+terraform {
+  required_providers {
+    scaleway = {
+      source  = "scaleway/scaleway"
+      version = "~> v2.52.0"
+    }
+  }
+}
+```
+
+## Migration steps
+
+### Legacy Configuration
+
+A typical legacy configuration might look like this:
+
+```hcl
+resource "scaleway_vpc" "main" {
+  name = "foo"
+}
+
+resource "scaleway_vpc_private_network" "main" {
+  name   = "bar"
+  vpc_id = scaleway_vpc.main.id
+}
+
+resource "scaleway_vpc_public_gateway_ip" "main" {
+}
+
+resource "scaleway_vpc_public_gateway" "main" {
+  name  = "foobar"
+  type  = "VPC-GW-S"
+  ip_id = scaleway_vpc_public_gateway_ip.main.id
+}
+
+resource "scaleway_vpc_public_gateway_dhcp" "main" {
+  subnet = "192.168.1.0/24"
+}
+
+resource "scaleway_instance_server" "main" {
+  image = "ubuntu_focal"
+  type  = "DEV1-S"
+}
+
+resource "scaleway_instance_private_nic" "main" {
+  server_id          = scaleway_instance_server.main.id
+  private_network_id = scaleway_vpc_private_network.main.id
+}
+
+resource "scaleway_vpc_gateway_network" "main" {
+  gateway_id         = scaleway_vpc_public_gateway.main.id
+  private_network_id = scaleway_vpc_private_network.main.id
+  dhcp_id            = scaleway_vpc_public_gateway_dhcp.main.id
+  cleanup_dhcp       = true
+  enable_masquerade  = true
+}
+
+resource "scaleway_vpc_public_gateway_dhcp_reservation" "main" {
+  gateway_network_id = scaleway_vpc_gateway_network.main.id
+  mac_address        = scaleway_instance_private_nic.main.mac_address
+  ip_address         = "192.168.1.1"
+}
+```
+
+### Triggering Migration
+
+Before updating your configuration, you must trigger the migration on the public gateway resource. For example, add the `migrate_to_v2` flag:
+
+```hcl
+resource "scaleway_vpc_public_gateway" "main" {
+  name          = "foobar"
+  type          = "VPC-GW-S"
+  ip_id         = scaleway_vpc_public_gateway_ip.main.id
+  migrate_to_v2 = true
+}
+```
+
+This call updates the gateway on the API side from v1 to v2. The DHCP configuration and reservations remain intact, but the underlying resource is now managed using v2.
```

Comment:
```suggestion
This call puts the gateway into IPAM mode and means it will now be managed by v2 of the API instead of v1. The DHCP configuration and reservations remain intact, but the underlying resource is now managed using v2.
```